climpyrical/tests/test_gridding.py

```diff
@@ -0,0 +1,301 @@
+from climpyrical.gridding import (
+    check_input_coords,
+    check_coords_are_flattened,
+    check_transform_coords_inputs,
+    flatten_coords,
+    transform_coords,
+    check_find_nearest_index_inputs,
+    check_find_element_wise_nearest_pos_inputs,
+    check_find_nearest_value_inputs,
+    find_nearest_index,
+    find_element_wise_nearest_pos,
+    find_nearest_index_value,
+)
+import pytest
+import numpy as np
+
+# create pseudo grids with expected dimension and ranges
+xi = np.linspace(-33.8800048828125, 33.8800048828125, 155)
+yi = np.linspace(-28.59999656677246, 28.15999984741211, 130)
+xext, yext = np.tile(xi, yi.size), np.repeat(yi, xi.shape)
+xext_bad, yext_bad = np.repeat(xi, yi.size), np.tile(yi, xi.shape)
```

Comment:
So in several places in this file you are defining global variables that are used by the methods underneath.  Instead of creating these variables here, it may be better for you to put them into `conftest.py` .  That way you they can just be passed as parameters rather than defined as globals and passed in from `mark.parametrize`.

```
# in conftest.py
@pytest.fixture(scope='function')
def xext_bad():
    # create the variable here
```

Reply:
Can you elaborate on why this would be better? Is `conftest.py` a standard practice for testing?

Reply:
I disagree in this particular case. The way you are using the global variables here is to construct parametrizations of test functions, and that much harder to do using fixtures. So for this particular use, to keep things simple, I wouldn't move these globals into `conftest.py` as fixtures.

To answer your broader question: `conftest.py` is core to `pytest`. It serves quite a range of test configuration, setup/teardown, and related purposes. It's where you put test fixtures, amongst other things, which are then automagically available globally to tests as function parameters. (Nikola's explanation conflates parametrization and fixture, which are not the same thing.)